sonar-batch/src/main/java/org/sonar/batch/issue/tracking/IssueTransition.java

```diff
@@ -66,9 +67,23 @@ public void execute() {
     }
 
     BatchReportReader reader = new BatchReportReader(reportPublisher.getReportDir());
+    int nbComponents = componentCache.all().size();
+
+    if (nbComponents == 0) {
+      return;
+    }
 
-    for (BatchComponent component : componentCache.all()) {
-      trackIssues(reader, component);
+    ProgressReport progressReport = new ProgressReport("issue-tracking-report", TimeUnit.SECONDS.toMillis(10));
+    progressReport.start("Performing issue tracking");
+    int count = 0;
+
+    try {
+      for (BatchComponent component : componentCache.all()) {
+        trackIssues(reader, component);
+        progressReport.message(++count + "/" + nbComponents + " components tracked");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)